### ISSUE TITLE
Hyperparameter tuning

### ISSUE THREAD

**Issue by GeorgeAyman1** [GeorgeAyman1/flight-delay-predictor#22]
**Tasks**

1. GridSearchCV / RandomSearch
2. Use validation set (2024)
